lib/rucio/core/rule.py

```diff
@@ -860,20 +993,26 @@ def list_rule_full_history(scope, name, *, session: "Session"):
     :raises:        RucioException
     """
 
-    query = session.query(models.ReplicationRuleHistory.id,
-                          models.ReplicationRuleHistory.created_at,
-                          models.ReplicationRuleHistory.updated_at,
-                          models.ReplicationRuleHistory.rse_expression,
-                          models.ReplicationRuleHistory.state,
-                          models.ReplicationRuleHistory.account,
-                          models.ReplicationRuleHistory.locks_ok_cnt,
-                          models.ReplicationRuleHistory.locks_stuck_cnt,
-                          models.ReplicationRuleHistory.locks_replicating_cnt).\
-        with_hint(models.ReplicationRuleHistory, "INDEX(RULES_HISTORY_SCOPENAME_IDX)", 'oracle').\
-        filter(models.ReplicationRuleHistory.scope == scope, models.ReplicationRuleHistory.name == name).\
-        order_by(models.ReplicationRuleHistory.created_at, models.ReplicationRuleHistory.updated_at)
-
-    for rule in query.yield_per(5):
+    stmt = select(
+        models.ReplicationRuleHistory.id,
+        models.ReplicationRuleHistory.created_at,
+        models.ReplicationRuleHistory.updated_at,
+        models.ReplicationRuleHistory.rse_expression,
+        models.ReplicationRuleHistory.state,
+        models.ReplicationRuleHistory.account,
+        models.ReplicationRuleHistory.locks_ok_cnt,
+        models.ReplicationRuleHistory.locks_stuck_cnt,
+        models.ReplicationRuleHistory.locks_replicating_cnt
+    ).with_hint(
+        models.ReplicationRuleHistory, "INDEX(RULES_HISTORY_SCOPENAME_IDX)", 'oracle'
```

Comment:
It never made sense to me that we have two quoting styles in the same line. Maybe it’s high time we fixed this? I also think we should use all-caps for SQL functions (this applies to other line like `index(CONTENTS CONTENTS_PK)`).

This comment applies throughout the file.

Reply:
Refactored the quoting style for `with_hint()` to use exclusively single quotes, like:

` with_hint(
        models.ReplicationRule, 'INDEX(RULES RULES_CHILD_RULE_ID_IDX)', 'oracle'`

`with_hint` Now also always using caps for SQL functions